### DISCUSSION TITLE
show help if validate fail

### DISCUSSION THREAD

**Discussion by AlessandroMenchiniTmaxLab** [spectreconsole/spectre.console#1451] [Q&A] [answered]
Hello everyone,

I was unable to find an example of argument validation that fits my needs. I need to define some CommandOptions as required and others as optional. In case the user does not enter a required option, I would like to show them the help for using the entire command. Is there a way to do this?
Regards

**Comment by FrankRay78** [comment, marked as answer]
>Is there a way to do this?

Yes, it's done out of the box by simply using spectre.console

Here's an example application with a single command and a mandatory argument:

```csharp
namespace ConsoleApp1
{
    public sealed class DogSettings : CommandSettings
    {
        [CommandArgument(0, "<AGE>")]
        public int Age { get; set; }
    }

    public class DefaultCommand : Command<DogSettings>
    {
        public override int Execute(CommandContext context, DogSettings settings)
        {
            AnsiConsole.WriteLine($"Dog age {settings.Age}");
            return 0;
        }
    }

    public class Program
    {
        static void Main(string[] args)
        {
            var app = new CommandApp<DefaultCommand>();
            app.Run(args);
        }
    }
}
```

I run it from the console by executing `ConsoleApp1.exe` and without the required argument, the help text is shown automatically:

![image](https://github.com/spectreconsole/spectre.console/assets/52075808/bfa5425b-5ec4-48ff-8765-6fbd11b77caa)

There are heaps of other examples located here, that you might find useful: https://github.com/spectreconsole/spectre.console/tree/main/examples
